### ISSUE TITLE
Documentation - clarifying basic concepts for writing own rules

### ISSUE THREAD

**Issue by ileben** [dictation-toolbox/Caster#837]
The current version of the documentation includes some misleading information as well as spreads some very important basic concepts throughout various pages. My recent discussion with Anatole Matveief on gitter shows this can be very confusing to new users. I think it would be useful to modify the documentation in the following way:

"Intro to rules and grammars"
- Remove any mention of grammars. Within Caster a grammar is an advanced concept the most people will never have to use. Due to the CCR system, even when writing custom rules, a user pretty much never has to worry about grammars. Including it on this page only serves to confuse the user and further reinforce the idea that using grammar.load (which is mentioned in other pages) is something that a user should normally do when using Caster.

"Dragonfly Rules"
- Move this whole section somewhere further down or deeper into a "migrating from dragonfly" section (I know a section like this is already in the works). Again I think, the information about loading of dragonfly rules is just unnecessary noise when someone is trying to learn how to use caster.

Instead of talking about loading of dragonfly rules I think it would be better to explain some very basic concepts really early on, possibly as part of "Intro to rules and grammars". This is what I wish someone explained to me when I first started writing my own rules:
"Normally, in lower level voice recognition libraries like Dragonfly only one command can be spoken in a single utterance. This means you have to pause your speech for a short moment before starting to speak a new command and after. The convenience of Caster's CCR system, is that it automatically merges dragonfly rules together so that you can speak multiple commands within the same utterance without pausing. The main advantage of this is that you don't have to pause between commands. The main disadvantage of this is that rules that end with a dictation element can get confused because they can't decide whether a word should be interpreted as the start of another command or as part of the dictation element. Built-in Caster rules work around this by including an optional terminator keyword in their grammar (eg "camel \<dictation\> [brunt]"). Caster supports both loading of merged rules (via the MergeRule class), as well as isolated top level rules (via the MappingRule class and CCRType.None). The details of this are explained in the "Loading rules" section [link]. The large majority of time you will want to utilize the MergeRule, but the option of top level rules is there should the need arise.

**Comment by LexiconCode** [comment]
Okay I've included many of the changes requested in this work in progress https://caster-lexiconcode.readthedocs.io/en/docs/

> "Intro to rules and grammars"
> 
>     * Remove any mention of grammars. Within Caster a grammar is an advanced concept the most people will never have to use. Due to the CCR system, even when writing custom rules, a user pretty much never has to worry about grammars. Including it on this page only serves to confuse the user and further reinforce the idea that using grammar.load (which is mentioned in other pages) is something that a user should normally do when using Caster.

I've clarified that individuals will be working rules and that caster manager grammars. More importantly the confusion stems from trying to teach Dragonfly. So I've simply removed all references to creating Dragonfly rules from the Rule Construction documentation. In doing so I've made some changes. Which leads to your comment about "Dragonfly Rules"


I moved all documentation relating to creating dragonfly grammars. In its place I've created Caster versions of those grammars and changed the order of how it's presented. Starting with their 1st rule, moving into basic rule then optionally advanced rules and finally loading rules.
